docs/blog/2019-04-01-setting-up-yarn-workspaces-for-theme-development/index.md

```diff
@@ -0,0 +1,186 @@
+---
+title: Setting up Yarn Workspaces for Theme Development
+date: 2019-04-01
+author: Brent Jackson
+excerpt: ""
+tags:
+  - themes
+  - yarn
+  - tutorials
+---
+
+Gatsby [themes][themes-tag] are a new way to share functionality across multiple Gatsby sites.
+Using [Yarn workspaces][] is an excellent way to set up a project for theme development.
+It allows you to keep multiple packages in a single parent directory and link dependencies together.
+For Gatsby theme development, that means you can keep multiple themes and example sites together in a single project.
+In this post, we'll walk through how to use Yarn workspaces to set up a development environment for creating custom themes.
+
+## Installation and directory structure
+
+First, if you don't have Yarn installed already, follow the directions here to [install Yarn][].
+Next, create a new directory for the entire project, where you'll be adding your theme and an example site later.
+
+Create two subdirectories in this folder: one for the theme itself and one for the example site.
+
+```shell
+mkdir gatsby-theme-example-workspaces example
+```
+
+Add a `package.json` to the root directory with these subdirectories in the `workspaces` field. Note that setting `"private": true` is required for Yarn workspaces.
+
+```json:title=package.json
+{
+  "private": true,
+  "workspaces": ["gatsby-theme-example-workspaces", "example"]
+}
+```
+
+Switch to each subdirectory and run `yarn init` to create a `package.json` for each one.
+Be sure the `name` field in your theme's `package.json` matches the directory name exactly.
+
+From the root directory, run the following to install Gatsby's dependencies for the example site.
+
+```shell
+yarn workspace example add gatsby react react-dom
+```
+
+The `yarn workspace` command will run Yarn commands for a specific workspace without needing to switch directories.
+
+Then add the following as peer dependencies to the theme.
+This will ensure that the end user of your theme can choose any compatible version of Gatsby.
+
+```shell
+yarn workspace gatsby-theme-example-workspaces add --peer gatsby react react-dom
+```
+
+## Base theme setup
+
+To ensure that Gatsby will compile ES6 syntax in your theme, add the following plugin to its `gatsby-config.js`.
+Note that this is only needed temporarily. Gatsby will automatically transpile themes in a later version.
+
+```shell
+yarn workspace gatsby-theme-example-workspaces add gatsby-plugin-compile-es6-packages
+```
+
+```js:title=gatsby-theme-example-workspaces/gatsby-config.js
+module.exports = {
+  plugins: [
+    {
+      resolve: "gatsby-plugin-compile-es6-packages",
+      options: {
+        modules: ["gatsby-theme-example-workspaces"],
+      },
+    },
+  ],
+}
+```
+
+Create an empty `index.js` file to the theme's directory that serves as a noop (empty function).
```

Comment:
Yeah, I don't have a strong opinion here – was mostly going off how this is described elsewhere. Pointing to the `gatsby-config.js` seems a *little* odd to me because it doesn't seem like a module that end users would need to import very often. But if this simplifies the steps in this post, then I think it makes sense